packages/catalog-api/src/lib/schema.graphql

```diff
@@ -28,13 +33,17 @@ type Mutation {
 }
 
 """
-Information about an npm package that's either version-independent
-or related to the latest version of the package.
+Information about an npm package that's independent of a specific version.
 """
-type PackageInfo {
+union PackageInfo = ReadablePackageInfo | UnreadablePackageInfo
+
+"""
+A package that has been successfuly imported
+"""
+type ReadablePackageInfo {
```

Comment:
This should not be possible. Readable means that the other fields have been filled out at least once and can be used to display data in a UI, even if they are stale. Because pakage versions are immutable, we can at least read something. I think even redacted packages will be a "readable" state, in that the data is there.

But given proper use of the Firestore document converters, if that case did happen (readable->unreadable) we would not return the additional fields of the readable type. This could be an area where we assert that database objects do not have extra fields.